docs/DesignDocs/CustomAttestation.md

```diff
@@ -0,0 +1,775 @@
+ 0xa1, 0xe9,                            \
+ 0x86, 0x2c, 0x62, 0x9b, 0x76, 0x2a}    \
+}
+
+/*! sgx_attestation_plugin
+ * 
+ * Return the SGX attesation plug-in.
+ */
+oe_attestation_plugin_t* sgx_attestation_plugin();
+
+```
+
+`sgx_attestation_plugin.c`
+
+```C
+#include "sgx_attestation_plugin.h"
+
+static
+oe_result_t
+sgx_attestation_plugin_on_register(
+    oe_attestation_plugin_t* plugin_context,
+    const void* config_data,
+    size_t config_data_size)
+{
+    OE_UNUSED(plugin_context);
+    OE_UNUSED(config_data);
+    OE_UNUSED(config_data_size);
+
+    // Nothing to do
+    return OE_OK;
+}
+
+static 
+oe_result_t 
+sgx_attestation_plugin_on_unregister(
+    oe_attestation_plugin_t* plugin_context)
+{
+    OE_UNUSED(plugin_context);
+
+    // Nothing to do
+    return OE_OK;
+}
+
+static 
+oe_result_t 
+sgx_attestation_plugin_get_evidence(
+    oe_attestation_plugin_t* plugin_context,
+    uint32_t flags,
+    const oe_claim_t* custom_claims,
+    size_t custom_claims_length,
+    const void* opt_params,
+    size_t opt_params_size,
+    uint8_t** evidence_buffer,
+    size_t* evidence_buffer_size,
+    uint8_t** endorsements_buffer,
+    size_t* endorsements_buffer_size)
+{
+    OE_UNUSED(plugin_context);
+
+    /*
+     * Pseudocode description instead of actual C code:
+     *
+     * Hash custom claims field.
+     * Call oe_get_report with the flags and opt_param parameters and the hash as reportdata.
+     * Report contains the endorsements, so extract them out.
+     * Evidence will be report + custom_claims blob.
+     *
+     * Note: Since the verifier can run outside the SGX enclave, it can be running on a
+     * machine with different endianness. Consequently, it must be possible for the verifier
+     * to determine the endianness of the multibyte numbers in the evidence and endorsements,
+     * so it can properly interpret them.
+     *
+     */
+
+    return OE_OK;
+}
+
+static 
+oe_result_t 
+sgx_attestation_plugin_free_evidence(
+    oe_attestation_plugin_t* plugin_context,
+    uint8_t* evidence_buffer)
+{
+    OE_UNUSED(plugin_context);
+
+    return oe_free_report(evidence_buffer);
+}
+
+static 
+oe_result_t 
+sgx_attestation_plugin_free_endorsements(
+    oe_attestation_plugin_t* plugin_context,
+    uint8_t* endorsements_buffer)
+{
+    OE_UNUSED(plugin_context);
+
+    return oe_free_endorsements(endorsements_buffer);
+}
+
+static 
+oe_result_t 
+sgx_attestation_plugin_verify_evidence(
+    oe_attestation_plugin_t* plugin_context,
+    const uint8_t* evidence_buffer,
+    size_t evidence_buffer_size,
+    const uint8_t* endorsements_buffer,
+    size_t endorsements_buffer_size,
+    const oe_datetime_t* input_validation_time,
+    oe_claim_t** claims,
+    size_t* claims_length)
+{
+    OE_UNUSED(plugin_context);
+
+    /*
+     * Pseudocode description instead of actual C code:
+     *
+     * Call oe_verify_report with all the input parameters and get the oe_identity_t back.
```

Comment:
Should this be extended to support validating against Intel vs Azure baseline?

Reply:
What do you mean Intel vs Azure baseline? The endorsements are part of the input parameters, so the caller can determine where to fetch it from

Reply:
I was referring to `oe_verify_report`. But if I understand you correctly, `oe_verify_report` would not look at the endorsements, it would be job of the `sgx_attestation_plugin_verify_evidence`?